components/scream/src/share/tests/common_physics_functions_tests.cpp

```diff
@@ -64,6 +64,79 @@ auto cmvdc (const ViewT& v_d) -> typename ViewT::HostMirror {
   return v_h;
 }
 
+template<typename DeviceT>
+void run_scalar_valued_fns()
+{
+  /*
+  Most common physics functions operate simultaneously on packs of vertical indices. The functions tested here
+  don't include any vertical dimension (e.g. they handle variables only defined at the surface), so are only defined
+  for scalar reals. This is fundamentally different than the other functions, so these functions get their own run
+  test.
+  */	
+
+  using RealType   = scream::Real;
+  using PF         = scream::PhysicsFunctions<DeviceT>;
+  using PC         = scream::physics::Constants<RealType>;
+  using Check = ChecksHelpers<RealType,1>; //1 is for number of levels.
+
+  static constexpr auto Rd       = PC::RD;
+  static constexpr auto g        = PC::gravit;
+  static constexpr auto test_tol = PC::macheps*1e3;
+
+  // lapse_T_for_psl property tests:
+  // If T_ground = 0, T_ground_tmp should be 255/2 and lapse should be 0.0065 Really cold case.
+  // If T_ground = 300 K with phi_ground>0 m2/s2, lapse = 0 and T_ground_tmp=0.5*(290.5+T_ground). Really hot case.
+  // If T_ground = 290 K and phi_ground=10000 m2/s2, T_ground_tmp=T_ground and lapse
+  //    is such that T_sl=T_ground+lapse*phi_ground/gravit is within roundoff of 290.5 K. Marginally hot case.
+  // If T_ground = 280 K and phi_ground=100 m2/s2, T_ground_tmp=T_ground and lapse=6.5 K/km (typical conditions)
+  RealType lapse;
+  RealType T_ground_tmp;
+  RealType T_ground=0;
+  RealType phi_ground=100;
+  RealType p_ground=100000; //can't use p0 because that's a pack
+  PF::lapse_T_for_psl(T_ground, p_ground,  phi_ground, lapse, T_ground_tmp );
+  REQUIRE( Check::approx_equal(T_ground_tmp,0.5*255.,test_tol) );
+  REQUIRE( Check::equal(lapse,0.0065) );
+  T_ground=300;
+  PF::lapse_T_for_psl(T_ground, p_ground,  phi_ground, lapse, T_ground_tmp );
+  REQUIRE( Check::approx_equal(T_ground_tmp,0.5*(290.5+T_ground),test_tol) );
+  REQUIRE( Check::equal(lapse,0) );
+  T_ground=290;
```

Comment:
Do you think it would add value to randomize the inputs? Perhaps over an interval that goes a tad beyond normal values (for robustness)?

Reply:
I definitely don't want to randomize the input for the property tests I've added here - I'm testing specific logic branches here. I wondered about *also* adding some randomized tests, but decided that they wouldn't add much value.